### ISSUE TITLE
Issue Label Notifications by using GitHub Action

### ISSUE THREAD

**Issue by ninginthecloud** [PyTorchLightning/pytorch-lightning#9216]
### Motivation

As PyTorch Lightning Community grows, some users want to be notified by issues with specific labels. By using GitHub Action would allow to notify the interested users. 
Basically, the GitHub Action runs every time a label is attached to an issue, and compares the label to the list of notification recipients that has been specified If a match is found, the action will add a comment to the issue and @mention the notification recipients.

### Pitch

Introduce Issue label notifier by using GitHub Action. 
https://github.com/marketplace/actions/issue-label-notifier

### Alternatives
Keep it as it is. 
Use the UI search tool to find the issues/PR with corresponding labels. 

### Additional context

Part of https://github.com/PyTorchLightning/pytorch-lightning/issues/9140

______________________________________________________________________

#### If you enjoy Lightning, check out our other projects! ⚡

<sub>

- [**Metrics**](https://github.com/PyTorchLightning/metrics): Machine learning metrics for distributed, scalable PyTorch applications.

- [**Flash**](https://github.com/PyTorchLightning/lightning-flash): The fastest way to get a Lightning baseline! A collection of tasks for fast prototyping, baselining, finetuning and solving problems with deep learning

- [**Bolts**](https://github.com/PyTorchLightning/lightning-bolts): Pretrained SOTA Deep Learning models, callbacks and more for research and production with PyTorch Lightning and PyTorch

- [**Lightning Transformers**](https://github.com/PyTorchLightning/lightning-transformers): Flexible interface for high performance research using SOTA Transformers leveraging Pytorch Lightning, Transformers, and Hydra.

</sub>

**Comment by justusschock** [comment]
If we introduce this, we also might want to introduce more fine grained labels w.r.t which part of the codebase it touches to directly ping the relevant people there.

**Comment by tchaton** [comment]
Yes, 

Let's re-work on our labels to ensure finer control on new issues.

Best,
T.C

**Comment by Borda** [comment]
It looks nice and seems to be kind of an alternative to PR code-ownsers...

**Comment by justusschock** [comment]
@Borda this is something different. Code Owners only requests reviews on PRs, this could ping people also on issues, which could allow faster resolution by directly pinging the right people :)

**Comment by Borda** [comment]
> @Borda this is something different. Code Owners only requests reviews on PRs, this could ping people also on issues, which could allow faster resolution by directly pinging the right people :)

yes that what I mean, the code-owners works only for PR, compare to this util would work repo-wide

**Comment by akihironitta** [comment]
I fancy this suggestion! Shall we proceed forward?
